python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+import csv
+from datetime import datetime
+import io
+import os
+import time
+
+import pytz
+import requests
+
+from columns import COLUMNS
+
+FILE_DIR = "cris_csvs"
+HASURA_ENDPOINT = "http://localhost:8084/v1/graphql"
+UPLOAD_BATCH_SIZE = 1000
+
+CRASH_UPSERT_MUTATION = """
+mutation UpsertCrashes($objects: [db_crashes_cris_insert_input!]!) {
+  insert_db_crashes_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: crashes_cris_pkey,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+UNIT_UPSERT_MUTATION = """
+mutation UpsertUnits($objects: [db_units_cris_insert_input!]!) {
+  insert_db_units_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_units_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}
+"""
+
+PERSON_UPSERT_MUTATION = """
+mutation UpsertPeople($objects: [db_people_cris_insert_input!]!) {
+  insert_db_people_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_people_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}"""
+
+mutations = {
+    "crashes": CRASH_UPSERT_MUTATION,
+    "units": UNIT_UPSERT_MUTATION,
+    "persons": PERSON_UPSERT_MUTATION,
+}
+
+
+def make_upsert_mutation(table_name):
+    """Sets the column names in the upsert on conflict array"""
+    upsert_mutation = mutations[table_name]
+    updateColumns = [
+        col["name"]
+        for col in COLUMNS
+        if col["table_name"] == table_name
+        and col["source"] == "cris"
+        and not col["is_primary_key"]
+    ]
+    return upsert_mutation.replace("$updateColumns", ", ".join(updateColumns))
+
+
+def make_hasura_request(*, query, endpoint, variables=None):
+    payload = {"query": query, "variables": variables}
+    res = requests.post(endpoint, json=payload)
+    res.raise_for_status()
+    data = res.json()
+    try:
+        return data["data"]
+    except KeyError:
+        raise ValueError(data)
+
+
+def get_file_meta(filename):
+    """
+    Returns:
+        schema_year, extract_id, table_name
+    """
+    filename_parts = filename.split("_")
+    return filename_parts[1], filename_parts[2], filename_parts[3]
+
+
+def get_files_todo():
+    files_todo = []
+    for filename in os.listdir(FILE_DIR):
+        if not filename.endswith(".csv"):
+            continue
+        schema_year, extract_id, table_name = get_file_meta(filename)
+
+        # ignore all tables except these
+        if table_name not in ("crash", "unit", "person", "primaryperson"):
```

Comment:
I'm looking forward to the conversation about what to do with the "tables" we get from CRIS but are not using at this time. I'd like to talk about the helpfulness of ingesting them, or at the very least, charges, since we regressed on that ages ago and this is a great chance to repair that.

Reply:
thanks for mentioning that. i just reached out to Xavier about it. i am resistant to bringing them into the new model if they're not in use today 🤷‍♂️

Reply:
i think that as long as we have a way to easily backfill and tables or columns in case they become relevant in the future, then we should be good but I'm open to hearing more! I also agree about charges and y'all's [convo about where to take them](https://austininnovation.slack.com/archives/CM9MK950S/p1712002389100689?thread_ts=1712001750.831599&cid=CM9MK950S). I think that is a good example of the benefit of using CRIS data but not shying away from molding it to fit our use cases. 💪